lib/ClangImporter/ClangModuleDependencyScanner.cpp

```diff
@@ -451,86 +451,98 @@ ClangImporter::getModuleDependencies(Identifier moduleName,
+    auto moduleCachePath = getModuleCachePathFromClang(getClangInstance());
+    auto lookupModuleOutput =
+        [moduleCachePath](const ModuleID &MID,
+                          ModuleOutputKind MOK) -> std::string {
+      return moduleCacheRelativeLookupModuleOutput(MID, MOK, moduleCachePath);
+    };
+    auto dependencies = clangScanningTool.getTranslationUnitDependencies(
+        commandLineArgs, workingDir, cache.getAlreadySeenClangModules(),
+        lookupModuleOutput);
+    if (!dependencies) {
+      // FIXME: Route this to a normal diagnostic.
+      llvm::logAllUnhandledErrors(dependencies.takeError(), llvm::errs());
+      return dependencies.takeError();
+    }
 
-  // Retrieve the bridging header.
-  std::string bridgingHeader = *(targetModule.getBridgingHeader());
+    // Record module dependencies for each new module we found.
+    auto bridgedDeps = bridgeClangModuleDependencies(
+        clangScanningTool, dependencies->ModuleGraph,
+        cache.getModuleOutputPath(),
+        [&cache](StringRef path) {
+          return cache.getScanService().remapPath(path);
+        });
+    cache.recordDependencies(bridgedDeps);
 
-  // Determine the command-line arguments for dependency scanning.
-  std::vector<std::string> commandLineArgs =
-      getClangDepScanningInvocationArguments(ctx, StringRef(bridgingHeader));
-  auto optionalWorkingDir = computeClangWorkingDirectory(commandLineArgs, ctx);
-  if (!optionalWorkingDir) {
-    ctx.Diags.diagnose(SourceLoc(), diag::clang_dependency_scan_error,
-                       "Missing '-working-directory' argument");
-    return true;
-  }
-  std::string workingDir = *optionalWorkingDir;
+    // Record dependencies for the source files the bridging header includes.
+    for (const auto &fileDep : dependencies->FileDeps)
+      targetModule.addHeaderSourceFile(fileDep);
 
-  auto moduleCachePath = getModuleCachePathFromClang(getClangInstance());
-  auto lookupModuleOutput =
-      [moduleCachePath](const ModuleID &MID,
-                        ModuleOutputKind MOK) -> std::string {
-    return moduleCacheRelativeLookupModuleOutput(MID, MOK, moduleCachePath);
+    // ... and all module dependencies.
+    llvm::StringSet<> alreadyAddedModules;
+    for (const auto &moduleDep : dependencies->ClangModuleDeps)
+      targetModule.addHeaderInputModuleDependency(moduleDep.ModuleName,
+                                                  alreadyAddedModules);
+
+    return dependencies;
   };
 
-  auto clangModuleDependencies =
-      clangScanningTool.getTranslationUnitDependencies(
-          commandLineArgs, workingDir, cache.getAlreadySeenClangModules(),
-          lookupModuleOutput);
-  if (!clangModuleDependencies) {
-    // FIXME: Route this to a normal diagnostic.
-    llvm::logAllUnhandledErrors(clangModuleDependencies.takeError(), llvm::errs());
-    return true;
+  // - Textual module dependencies require us to process their bridging header.
```

Comment:
Can textual module has bridging header?

Reply:
This came from existing code which always includes textual modules in querying this. I am not actually sure, that's worth revisiting. But as-is, the code handles source *and* textual dependencies in all of the bridging header handling code. If we're looking to change that, I'd like that to be a separate change.

Reply:
I don't think you can actually formulate a compilation with bridging header and make it generate interface file now. But sure, if you want to delete the code later, I am fine with that.